src/components/header/header.tsx

```diff
@@ -1,42 +1,47 @@
 'use client';
+
 import { Box, Link, Tab, Tabs } from '@mui/material';
-import React from 'react';
+import { usePathname } from 'next/navigation';
+import React, { useEffect, useState } from 'react';
+import HeaderEnum from '../../app/dataModels/enum/header-enum';
+import { headerData, pathToEnumMap } from '../../utils/maps/header-map';
```

Comment:
I am confused why you need to import headerData from here.

Also, instead of importing pathToEnumMap, why not export the function generateHeaderData. Use the map directly in your component so you can avoid a unnecessary variable export and decoration?